ipfs/publish.go

```diff
@@ -30,3 +44,107 @@ func Publish(ctx commands.Context, hash string) (string, error) {
 	log.Infof("Published %s to IPNS", hash)
 	return returnedVal, nil
 }
+
+// Publish another IPFS record at /ipns/<peerID>:<altRoot>
+func PublishAltRoot(ctx commands.Context, nd *core.IpfsNode, altRoot string, value path.Path, eol time.Time) error {
+	hash, err := mh.FromB58String(nd.Identity.Pretty())
+	if err != nil {
+		return err
+	}
+	ipnskey := "/ipns/" + string(hash) + ":" + altRoot
+
+	// get previous records sequence number
+	seqnum, err := getPreviousSeqNo(context.Background(), nd, ipnskey)
+	if err != nil {
+		return err
+	}
+
+	// increment it
+	seqnum++
+
+	return PutRecordToRouting(context.Background(), ipnskey, nd.PrivateKey, value, seqnum, eol, nd.Routing)
+}
+
+func getPreviousSeqNo(ctx context.Context, nd *core.IpfsNode, ipnskey string) (uint64, error) {
+	prevrec, err := nd.Repo.Datastore().Get(dshelp.NewKeyFromBinary([]byte(ipnskey)))
+	if err != nil && err != ds.ErrNotFound {
+		// None found, lets start at zero!
+		return 0, err
```

Comment:
It looks like the condition `err != ds.ErrNotFound` and your comment underneath disagree with each other.

Reply:
Yeah that looks correct. Chris wrote this and so I'm not sure but it definitely looks like a bug. I will address this.